### PR TITLE
Add documentation for Q-functions and some missing details in docstrings

### PR DESCRIPTION
**Pull request by marioyc**

### CODE REVIEW

File: chainerrl/q_functions/state_q_functions.py [hunk 1]
```diff
@@ -160,6 +160,8 @@ class FCQuadraticStateQFunction(
         chainer.Chain, StateQFunction):
     """Fully-connected state-input continuous Q-function.
 
+    See: http://arxiv.org/abs/1603.00748
```
Comment (toslunar) [inline chainerrl/q_functions/state_q_functions.py]: `https` looks better to me.

File: chainerrl/q_function.py [hunk 2]
```diff
@@ -13,14 +13,27 @@
...
         raise NotImplementedError()
 
 
 class StateActionQFunction(with_metaclass(ABCMeta, object)):
+    """Abstract Q-function with state and action input."""
 
     @abstractmethod
     def __call__(self, x, a):
+        """Evaluates Q-function
+
```
Comment (toslunar) [inline chainerrl/q_function.py]: Could you add `Args:` section?

File: chainerrl/q_functions/state_q_functions.py (hunk 1 above)
Comment (marioyc) [reply chainerrl/q_functions/state_q_functions.py]: Done

File: chainerrl/q_function.py (hunk 2 above)
Comment (marioyc) [reply chainerrl/q_function.py]: Done

**Pull request merged**